### DISCUSSION TITLE
Cache invalidation tool

### DISCUSSION THREAD

**Discussion by sumantapakira** [Adobe-Consulting-Services/acs-aem-commons#3493] [General]
I think every project uses CDN and there is a need to invalidate cache by authors. Have you already though about creating such tool? If not, then I can create PR.

**Comment by kwin** [comment]
Usually relying on TTLs set via Cache-Control headers is good enough. Explicit invalidation should be triggered from outside AEM and there are already different endpoints (outside AEM) for the different CDNs for that:
a) Fastly CDN shipping with AEMaaCS: https://experienceleague.adobe.com/en/docs/experience-manager-learn/cloud-service/caching/how-to/purge-cache
b) Akamai e.g. https://github.com/akamai/cli-purge

**Reply by sumantapakira** [reply]
Yes correct, this option is good when TTL is short but when we set longer TTL then author need a way to invalidate cache as soon as they publish contents. So such cases they need to depends on developer to execute purge request. wasting of time. how do you resolve this? Always set TTL short?
